tests/BenchmarkDotNet.Tests/Validators/CompilationValidatorTests.cs

```diff
@@ -59,6 +60,19 @@ public void BenchmarkedMethodNameMustNotUseCsharpKeywords()
                     "Benchmarked method `typeof` contains illegal character(s) or uses C# keyword. Please use `[<Benchmark(Description = \"Custom name\")>]` to set custom display name."));
         }
 
+        [Theory]
+        /* BenchmarkDotNet can only benchmark public methods */
+        [InlineData(typeof(SealedClass), true)]
+        [InlineData(typeof(MyPrivateClass), true)]
+        [InlineData(typeof(BenchMarkPublicClass), false)]
+        /* Generics Remaining */
+        public void Benchmark_Class_Modifers_Must_Be_Public(Type type, bool hasErrors)
```

Comment:
Maybe redundant, but it might be good to add tests for `internal` and `protected` also.

Reply:
I first implemented them, but public/private access were checked first, So I too thought that they were redundant and removed them